tapdance/dialer.go

```diff
@@ -99,16 +97,9 @@ func (d *Dialer) DialContext(ctx context.Context, network, address string) (net.
 		var cjSession *ConjureSession
 
 		transport := d.TransportConfig
```

Comment:
I think this commented out block needs to be uncommented so that we can support older clients using the conjure `Dialer` without forcing an API change upon them. You might also add a subsequent check, like if after `ConfigFromTransportType` is called `transport` is still `nil` then we return an error.

Reply:
Also we should check the error returned by `ConfigFromTransportType`.